### DISCUSSION TITLE
how to set username/password for GUI

### DISCUSSION THREAD

**Discussion by aamirismail** [haugene/docker-transmission-openvpn#1986] [Q&A] [answered]
In the original transmission image you can pass in enviorment variable for GUI username/password. 
How to do in this image?  I feel its a security risk as transmission vpn is reachable via internet on 9091 without any credentials.

 - USER=#optional
 - PASS=#optional

**Comment by pkishino** [comment, marked as answer]
Read the documentation
http://haugene.github.io/docker-transmission-openvpn/config-options/
It’s called rpc authentication, check the link to the official list of transmission variables

**Reply by pkishino** [reply]
http://haugene.github.io/docker-transmission-openvpn/faq/

**Comment by PepinoMartinez** [comment]
After using the above recommendation and setting authorization, I have the following 2 problems (never before):
- most times (like 9/10) the sequence to assign an open port (from PIA) to transmission seems to start but not working at last. Transmission has a port that is not open.
- Portainer has the following log error just after initializing the sequence to assign a port to transmission: `Unexpected response: <h1>401: Unauthorized</h1>` and keeps logging this error.

After removing the authorization (- TRANSMISSION_RPC_AUTHENTICATION_REQUIRED=false), everything returns to normal.
